### DISCUSSION TITLE
Contributor call: 29 April 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#3823] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: April 29 2023 **4pm** UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/apr-29-2023/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda
- Can we stop sending end-users to the lab please :pray: 
- When joost is being stupid again please tell him
- Apply [the Diataxis framework](https://diataxis.fr/) to the FreeSewing.org docs (as on FreeSewing.dev) yay or nay?
- Do we need/want a documentation team? What would such a team do? Who would (like to) be in it?
- Should we keep v2 bugs open if the fix is in v3?
- plugin prefixes and the general approach to *undoing* macros (see #3841)
- The tone we want in sewing related documentation and personal language in documentation.

**Comment by SeaZeeZee** [comment]
I would like to start a documentation team to take that work off of Joost's plate.

**Reply by BenJamesBen** [reply]
I think that having a documentation team is a good idea. Documentation should be treated as a first-class discipline/activity similar to pattern design and development.

**Reply by joostdecock** [reply]
I've added it to the agenda. Although I feel compelled to point out that while I don't disagree, we don't have a pattern design team, or development team either :shrug:

**Reply by BenJamesBen** [reply]
? I know that we don't have official/formal "teams". However, when I look at Discord I see a "Contributor Teams" menu with "pattern-design" and "development" channels under it. I thought that meant that there is a pattern design team and also a development team?

**Reply by SeaZeeZee** [reply]
I am not going to be available for the call but I do want to give my thoughts so I'll do it here. At the moment some of the main people that are involved in documentation like @bobgeorgethe3rd and @tangerineshark don't have write access to the monorepro. What I mean by a documentation team is mainly that perhaps those people should have write access. And with it more space to handle documentation so that for example Joost doesn't have to handle it unless he wants to. Another aspect of this is that occasionally in the past things have been merged while someone was still working on review and so documentation (and all of this is mainly sewing documentation) got merged without a review for clarity ect. Is it possible for PRs to be blocked from merging unless reviews have gone through?

**Reply by starfetch** [reply]
Will not be able to attend the call today as well, just here to say that I'm also interested in joining the docs team.

**Comment by BenJamesBen** [comment]
- For bugs that exist in v2 but that have fixes already merged into the v3 codebase, should we close the GitHub Issues for those bugs now, before v3 is deployed and available to the general public?

**Reply by joostdecock** [reply]
Added to the agenda

**Comment by bobgeorgethe3rd** [comment]
https://github.com/freesewing/freesewing/pull/3841
So due to working on this PR it dawned on me there is an argument to be had to standardise this behaviour across all of the plugin-annonations that are macros so they all have `prefix` functionality and others behaviours that can be across the board.

**Reply by joostdecock** [reply]
Added to the agenda

**Comment by BenJamesBen** [comment]
The tone we want in sewing related documentation and personal language in documentation.
https://github.com/freesewing/freesewing/pull/3842
https://github.com/freesewing/freesewing/issues/3750

**Reply by joostdecock** [reply]
Added to the agenda

**Comment by marpants** [comment]
If there's time I'd like to get to know the contributors a bit better. I really appreciate the community and would love to know a little more about the people who are working so hard to support it, for example, what drives you to dedicate so much of your free time? Do you have any specific long term desires for the site? Idk how comfortable people feel about it though, or if this is the right place 🙆

**Comment by marpants** [comment]
Also, ideas for collecting maker feedback on docs/patterns so we can use it as a reference when making related changes.

**Comment by marpants** [comment]
OH one more, the [freesewing youtube](https://www.youtube.com/@freesewing2874/): is it dead? Was/is there any intention for contributors to post videos there? Can/should we encourage people to make/post videos there? I'm planning to make a video for the Charlie Fly constructions since the images just can't quite do it justice, and the reference video isn't very beginner friendly.

**Comment by tangerineshark** [comment, marked as answer]
### Notes
We did some intros again! Probably good to do these semi-regularly. Welcome to new attendees @marpants and Thrunic

Can we stop sending end-users to the lab please 🙏

- This one's easy, just remember that for new users of FreeSewing, the lab might be more confusing than what they're looking for, so for end-users, try to avoid sending them to the lab unless it's necessary

Apply [the Diataxis framework](https://diataxis.fr/) to the FreeSewing.org docs (as on FreeSewing.dev) yay or nay?

- Survey says nay
- @joostdecock shares that .dev site uses diataxis framework for site organization and wonders whether it should be applied to the .org site organization as well
- @BenJamesBen feels that it's a challenging way to organize things if you don't know what documentation exists already
- @eriese mentions that docs can be a bit confusing re: the difference between a guide and a how-to, for instance, and suggests that non-design-specific docs (e.g. sewing a fly) go one place, while get-involved-with-FreeSewing docs are a third category
- **So, a note for everyone, feedback is requested on how you would like the URLs for docs to be structured.**

Do we need/want a documentation team? What would such a team do? Who would (like to) be in it?

- @bobgeorgethe3rd suggestion to add labels, @joostdecock suggests a proposal for how to use labels in github / for issues, this has since been documented way better as a discussion here: https://github.com/freesewing/freesewing/discussions/3952 
- do we also want a documentation Discord channel?
- @bobgeorgethe3rd : add maintainers to front matter for documentation pages so it's easier to check in with them
- @BenJamesBen : you can also submit a PR as a draft, allowing folks to note that things aren't ready to be merged yet
- @marpants : is there a way to collect feedback from the website and/or Discord?

Other documentation notes

- General consensus is that documentation is a little difficult to find and navigate
- @marpants suggests that it sounds like we need a proposal on how the docs should be structured
- @woutervdub mentions that multiple URLs can point to the same address

Should we keep v2 bugs open if the fix is in v3?

- We can close v2 bugs that are broken in v2 but fixed in v3

plugin prefixes and the general approach to undoing macros (see https://github.com/freesewing/freesewing/pull/3841)

- @bobgeorgethe3rd  has been working on a PR to deal with undoing certain macros (like, let's say you extend a pattern and you want to remove some dimension or whatnot) - @joostdecock shares that it's a good PR, and perhaps we should have a more consistent/reliable way to do this, so he's gonna put forward a proposal

Last but not least @marpants asked about the current FreeSewing YouTube and has since been added as a manager 🙌

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
